Introduction to DEA Coastlines notebook

robbibt opened the pull request:
### Proposed changes
This notebook provides a basic introduction to DEA Coastlines, including how to load data using WFS, and how to analyse the data to extract transect plots and histograms of change.

This was written a month or so ago, but after seeing @MatthewJA's DEA Waterbodies notebook, I think I may revisit this one in a week or so to simplify it to just covering data load and plotting, and move the more advanced functions to a `Real_world_example` notebook. However, as we need this online so we can forward it to stakeholders to demonstrate how to load data, I might leave this for a future task.

This PR also involves changes to simplify the `dea_coastaltools` and `dea_plotting` scripts, so I have re-run the `Intertidal_elevation` and `Coastal_erosion` notebooks to account for those changes, and updated those notebooks to cross-reference DEA Coastlines now that it exists.

### Checklist (replace `[ ]` with `[x]` to check off)
- [x] Notebook created using the [DEA-notebooks template](https://github.com/GeoscienceAustralia/dea-notebooks/tree/develop)
- [x] Remove any unused Python packages from `Load packages`
- [x] Remove any unused/empty code cells
- [x] Remove any guidance cells (e.g. `General advice`)
- [x] Ensure that all code cells follow the [PEP8 standard](https://www.python.org/dev/peps/pep-0008/) for code. The `jupyterlab_code_formatter` tool can be used to format code cells to a consistent style: select each code cell, then click `Edit` and then one of the `Apply X Formatter` options (`YAPF` or `Black` are recommended).
- [x] Include relevant tags in the final notebook cell (refer to the [DEA Tags Index](https://docs.dea.ga.gov.au/genindex.html), and re-use tags if possible)
- [x] Clear all outputs, run notebook from start to finish, and save the notebook in the state where all cells have been sequentially evaluated
- [x] Test notebook on both the `NCI` and `DEA Sandbox` (flag if not working as part of PR and ask for help to solve if needed)
- [x] If applicable, update the `Notebook currently compatible with the NCI|DEA Sandbox environment only` line below the notebook title to reflect the environments the notebook is compatible with

robbibt committed:
Add DEA Coastlines notebook and code, as well as metadata func

robbibt committed:
Fix invalid html tags issue

robbibt committed:
Run notebook from top

robbibt committed:
Add notebook to readme

robbibt committed:
Update existing coastal notebooks to better reflect DEA Coastlines code

robbibt committed:
Minor markdown update

robbibt committed:
Add link to sandbox to erosion notebooks

robbibt committed:
Remove metadata load script for time being

robbibt committed:
Merge develop

robbibt committed:
Re-run all cells

robbibt committed:
Fix duplicated study area in intertidal notebook

BexDunn reviewed (changes requested):
Hey Robbi, can you please caption the 3 images in DEA_Coastlines.ipynb with what the legend means/ the colours/ what's going on? I think it's a coastline...

BexDunn commented:
I think the Coastal Erosion notebook is fantastic - only 2 points, first one being that compatibility has an I in the middle not an a, and that there are two images that a caption would help me to interpret :)

MatthewJA commented:
It'd be nice to use a WFS library instead of writing your own WFS GET request. We used owslib in dea_waterbodies.py. That said, this is kind of a pain, so maybe we can flag this in an issue if you agree and get to it later.

MatthewJA commented on DEA_datasets/DEA_Coastlines.ipynb:
```diff
@@ -0,0 +1,1225 @@
+{
```
<p>Axis labels please! Cool plot though</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GeoscienceAustralia/dea-notebooks/pull/698/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

MatthewJA replied:
<p>GIS -&gt; GIS program?</p><p> </p><p>I have never heard the noun "transect" before, so maybe consider a more well-known term here?</p><p> </p><p>There is a 50 km limitation in how long the transect can be - you should say this in the cell</p><p> </p><p>This widget is super cool, but I can't get any outputs that aren't errors! Something is unclear here :(</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GeoscienceAustralia/dea-notebooks/pull/698/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

MatthewJA replied:
<p>Update the date :)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GeoscienceAustralia/dea-notebooks/pull/698/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

MatthewJA commented on Scripts/dea_spatialtools.py:
```diff
@@ -419,9 +419,16 @@ def contours_to_multiline(da_i, z_value, min_vertices=2):
+                             for i in find_contours(da_i.data, z_value) 
+                             if i.shape[0] > min_vertices]
+            
+        except:
```
Naked exceptions aren't great - please choose a specific exception! It'll also be easier to read - I'm not sure what you're trying to catch here.

robbibt committed:
Update for review comments

robbibt committed:
Remove try/except and add note about scikit-image issue

robbibt committed:
Add additonal markdown and instructions for exporting data

robbibt committed:
Add screenshots

robbibt commented:
Hi all, thanks for the great comments. @BexDunn : I've split the first plot into two and used it as a nice place to explain both datasets, as well as add in a legend and titles. I've also added captions under the two plots generated by the widgets to better explain what they show. Can you point me to the two figures in `Coastal_erosion` you'd like captions for too?

@MatthewJA : have updated all the minor things, for the more specific ones:

> I have never heard the noun "transect" before, so maybe consider a more well-known term here?

Transect is well-used as a noun in coastal monitoring applications, as well as others in ecology (e.g. see the [second definition here](https://dictionary.cambridge.org/dictionary/english/transect): `a line or narrow area along or within which measurements are taken, and items counted, etc. in scientific studies`). I've left it here for now

> This widget is super cool, but I can't get any outputs that aren't errors! Something is unclear here :(

I've added screenshot examples for the two widgets - can you have another go and see if it's more clear? I'd love to make it as user friendly for new users as possible

> Naked exceptions aren't great - please choose a specific exception! It'll also be easier to read - I'm not sure what you're trying to catch here.

This was code I internally used in DEA Coastlines to catch an obscure `scikit-image` bug (https://github.com/scikit-image/scikit-image/issues/4830) that is yet to be fixed. It probably isn't needed in `dea-notebooks`, so I have removed it for now and added a note about the bug in case others encounter it.

robbibt commented:
Missed this one:

> It'd be nice to use a WFS library instead of writing your own WFS GET request. We used owslib in dea_waterbodies.py. That said, this is kind of a pain, so maybe we can flag this in an issue if you agree and get to it later.

I actually like the simplicity of the WFS GET request as it requires only one simple dependency (`geopandas`). That said, I will probably re-visit this when I wrap some of the loading code into funcs like in `dea_waterbodies.py`.

robbibt committed:
Reduce size of screenshots

MatthewJA reviewed (approved):
Sounds good.

BexDunn reviewed (approved):
Looks spectacular
